generator/config/generator/tsl_templates/core/header_file.template

```diff
@@ -6,6 +6,10 @@
 #ifndef {{ tsl_include_guard }}
 #define {{ tsl_include_guard }}
 
+{% for preliminary_define in preliminary_defines %}
```

Comment:
Please insert a check whether preliminary_defines exists, otherwise the twig conversion will fail.